### DISCUSSION TITLE
flet pack with multiple scripts

### DISCUSSION THREAD

**Discussion by Kawue** [flet-dev/flet#4489] [Q&A]
**Description**
My flet app is splitted between multiple scripts. One script that contains the gui code, one that contains data loading, one that contains processing, etc.

If I use `flet pack` on the `gui.py` script my own scripts are ignored which breaks the imports. Can I use `flet pack` on a whole folder to keep my code separated?

**Comment by willjejones** [comment]
I've run into the same problem with my project. It's possible we've both missed something but I couldn't see any way to `flet pack` multiple files - this should really be supported if it isn't already!

It may be a bit late, but I thought I'd share my workaround. I used [stickytape](https://github.com/mwilliamson/stickytape) to convert my main file (e.g. your `gui.py` script) and all of it's dependencies (local modules and packages) into a single-file Python script. I then run `flet pack` on this new file, which seems to work great.

This is a totally hacky solution, but given my project is a hacky/reasearch one anyway, I wasn't too bothered.

Hope this helps someone else reading, and I hope `flet` considers supporting packing directories if this is not already implemented or in the roadmap.

**Comment by Mayurji** [comment]
@ndonkoHenri, Sorry to add you here, Had a question on "How to pack multiple files in flet?"

**Comment by ndonkoHenri** [comment]
Maybe use the `--add-data` and/or `--hidden-import` params ?
